### PR TITLE
Reset index

### PR DESCRIPTION
**Pull request by nagrgkgc**
I have added index status reset of VSINDEX.
you can verify and merge.

### CODE REVIEW

File: src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    /**
+     * RebuildEsIndexCommand constructor.
+     *
+     * @param  ObjectManagerFactory  $objectManagerFactory
+     * @param  ManagerInterface  $eventManager
+     * @param  array  $excludeIndices
+     */
+    public function __construct(
+        ObjectManagerFactory $objectManagerFactory,
+        ManagerInterface $eventManager
```
Comment (afirlejczyk) [inline src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php:33]: this dependency `ManagerInterface` is not used, it should be removed from the constructor.

File: src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                        ->save();
+		    $output->writeln("\n" . $indexer->getTitle() . "\n");
+                } catch (LocalizedException $e) {
+                    //catch exception
+                    $output->writeln("<error>" . $e->getMessage() . "</error>");
+                }
+            }
+        }
+
+        return $invalid;
```
Comment (afirlejczyk) [inline src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php:79]: this action shouldn't really return anything, no need to return an empty array. please remove return statement and `invalid = [];` from line 63

File: src/module-vsbridge-indexer-core/Console/Command/RebuildEsIndexCommand.php [hunk 3]
```diff
@@ -121,7 +121,7 @@ protected function execute(InputInterface $input, OutputInterface $output)
 
         if (!empty($invalidIndices)) {
             $message = 'Some indices has invalid status: '. implode(', ', $invalidIndices) . '. ';
-            $message .= 'Please change indices status to VALID manually.';
+            $message .= 'Please change indices status to VALID manually or php bin/magento vsbridge:resetindex.';
```
Comment (afirlejczyk) [inline src/module-vsbridge-indexer-core/Console/Command/RebuildEsIndexCommand.php:124]: change text to `or run php bin/magento vsbridge:resetindex` or `or use php bin/magento vsbridge:resetindex`

File: src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        parent::configure();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        //invalidate all indexes
+        $invalidIndices = $this->getInvalidIndices();
```
Comment (afirlejczyk) [inline src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php:55]: as mention below, `getInvalidIndices` shouldn't return anything.
change this to `$this->getInvalidIndices();`

File: src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    private function getIndexers()
+    {
+        /** @var IndexerInterface[] */
+        $indexers = $this->getAllIndexers();
+        $vsbridgeIndexers = [];
+
+        foreach ($indexers as $indexer) {
+            $indexId = $indexer->getId();
+
+            if (substr($indexId, 0, 9) === 'vsbridge_') {
```
Comment (afirlejczyk) [inline src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php:94]: I think we could assign this `vsbridge_` to const

File: src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    {
+        $invalid = [];
+
+        foreach ($this->getIndexers() as $indexer) {
+            if ($indexer->isWorking()) {
+                try {
+                    $indexer->getState()
+                        ->setStatus(\Magento\Framework\Indexer\StateInterface::STATUS_INVALID)
+                        ->save();
+		    $output->writeln("\n" . $indexer->getTitle() . "\n");
```
Comment (afirlejczyk) [inline src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php:71]: Maybe we should change it to 
` $output->writeln($indexer->getTitle() . ' indexer has been invalidated.');`
To have the same message like in Magento
https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Indexer/Console/Command/IndexerResetStateCommand.php

File: src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+use Symfony\Component\Console\Output\OutputInterface;
+use Magento\Framework\Exception\LocalizedException;
+
+/**
+ * Class IndexerReindexCommand
+ */
+class ResetEsIndexCommand extends AbstractIndexerCommand
+{
+    /**
+     * RebuildEsIndexCommand constructor.
```
Comment (afirlejczyk) [inline src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php:25]: change it to `ResetEsIndexCommand constructor`

File: src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+use Magento\Framework\App\ObjectManagerFactory;
+use Magento\Framework\Event\ManagerInterface;
+use Magento\Framework\Indexer\IndexerInterface;
+use Magento\Indexer\Console\Command\AbstractIndexerCommand;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Magento\Framework\Exception\LocalizedException;
+
+/**
+ * Class IndexerReindexCommand
```
Comment (afirlejczyk) [inline src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php:20]: wrong class name

File: src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php [hunk 9]
```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * @package  Divante\VsbridgeIndexerCore
+ * @author Nagaraja Kharvi <nagrgk@gmail.com>
+ * @copyright 2019 Divante Sp. z o.o.
+ * @license See LICENSE_DIVANTE.txt for license details.
```
Comment (afirlejczyk) [inline src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php:6]: Recently I start to simplify file headers, please change it to:
```
/**
 * Copyright Divante Sp. z o.o.
 * See LICENSE_DIVANTE.txt for license details.
 */
```

**Comment by afirlejczyk** [comment]
Thanks for your PR. After changes will be made I will merge it.

**Comment by afirlejczyk** [comment]
Closing in favour of https://github.com/DivanteLtd/magento2-vsbridge-indexer/pull/303.

File: src/module-vsbridge-indexer-core/Console/Command/RebuildEsIndexCommand.php (hunk 3 above)
Comment (nagrgkgc) [reply src/module-vsbridge-indexer-core/Console/Command/RebuildEsIndexCommand.php:124]: done

File: src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php (hunk 9 above)
Comment (nagrgkgc) [reply src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php:6]: done

File: src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php (hunk 8 above)
Comment (nagrgkgc) [reply src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php:20]: previous commit, done

File: src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php (hunk 1 above)
Comment (nagrgkgc) [reply src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php:33]: removed

File: src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php (hunk 4 above)
Comment (nagrgkgc) [reply src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php:55]: done

File: src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php (hunk 6 above)
Comment (nagrgkgc) [reply src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php:71]: done

File: src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php (hunk 2 above)
Comment (nagrgkgc) [reply src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php:79]: done

File: src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php (hunk 1 above)
Comment (nagrgkgc) [reply src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php:33]: done

File: src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php (hunk 2 above)
Comment (nagrgkgc) [reply src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php:79]: done

File: src/module-vsbridge-indexer-core/Console/Command/RebuildEsIndexCommand.php (hunk 3 above)
Comment (nagrgkgc) [reply src/module-vsbridge-indexer-core/Console/Command/RebuildEsIndexCommand.php:124]: done

File: src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php (hunk 4 above)
Comment (nagrgkgc) [reply src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php:55]: done

File: src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php (hunk 5 above)
Comment (nagrgkgc) [reply src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php:94]: done

File: src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php (hunk 6 above)
Comment (nagrgkgc) [reply src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php:71]: done

File: src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php (hunk 7 above)
Comment (nagrgkgc) [reply src/module-vsbridge-indexer-core/Console/Command/ResetEsIndexCommand.php:25]: done

**Pull request closed without merging**